### ISSUE TITLE
Edge cases in empty-lines rule

### ISSUE THREAD

**Issue by gsamokovarov** [mgechev/revive#82]
**Describe the bug**

The following example fails:

```go
func f10(*int) bool {
	if x > 2 {
		return true
	}
}
```

This is because of the trailing rule taking the position if the `if` statement, not the closing of the block into account.

**Expected behavior**

This is a false-positive and should not happen.
